include/slang/binding/OperatorExpressions.h

```diff
@@ -276,15 +291,21 @@ class StreamingConcatenationExpression : public Expression {
 
     template<typename TVisitor>
     void visitExprs(TVisitor&& visitor) const {
-        for (auto op : streams())
-            op->visit(visitor);
+        for (auto stream : streams()) {
+            stream->operand->visit(visitor);
+            if (stream->with) {
+                stream->with->left->visit(visitor);
+                if (stream->with->right)
+                    stream->with->right->visit(visitor);
+            }
+        }
     }
 
     bool isFixedSize() const;
     size_t bitstreamWidth() const;
 
 private:
-    span<const Expression* const> streams_;
+    span<const StreamExpression* const> streams_;
```

Comment:
Same here, you can just store these directly in the list instead of indirectly through a pointer.

Reply:
I can make the change. However, StreamExpression is currently allocated by the bump allocator
```c++
      buffer.append(compilation.emplace<StreamExpression>(StreamExpression{ arg, with }));
```
If I make the change, it will be heap allocated at
```c++
    return *compilation.emplace<StreamingConcatenationExpression>(
```
Can you confirm that it really improves allocation?